pinc/stages.inc

```diff
@@ -308,15 +304,13 @@ function can_user_get_pages_in_project($username, $project, $round)
                 ");
                 $row = mysqli_fetch_assoc($result);
                 if ($row["pagesdone"] >= $max_n_pages_per_project) {
-                    return
-                        _("You have reached your quota of pages from this 'Beginners Only' project, but you're welcome to work on a different one.")
-                        . " "
-                        . _("Or you could try working on an EASY project.");
+                    throw new BeginnersProjectQuotaReachedException(_("You have reached your quota of pages from this 'Beginners Only' project, but you're welcome to work on a different one. Or you could try working on an EASY project."));
                 }
             }
         } elseif ($round->is_a_mentor_round()) {
             if (!user_can_work_on_beginner_pages_in_round($round)) {
-                return _("You do not have access to difficulty='beginner' (Mentors Only) projects in this round.");
+                $message = _("You do not have access to difficulty='beginner' (Mentors Only) projects in this round.");
```

Comment:
Either assign `$message` and use it, or just put it in the exception like you do above. Probably the latter.

Reply:
oops.